etc/config/c++.amazon.properties

```diff
@@ -771,9 +771,21 @@ compiler.icc202160.libPath=/opt/compiler-explorer/intel-cpp-2022.1.0.137/compile
 compiler.icc202160.semver=2021.6.0
 compiler.icc202160.options=-gxx-name=/opt/compiler-explorer/gcc-10.1.0/bin/g++
 
+compiler.icc202170.exe=/opt/compiler-explorer/intel-cpp-2022.2.0.8772/compiler/latest/linux/bin/intel64/icc
+compiler.icc202170.ldPath=/opt/compiler-explorer/intel-cpp-2022.2.0.8772/compiler/latest/linux/compiler/lib/intel64_lin
+compiler.icc202170.libPath=/opt/compiler-explorer/intel-cpp-2022.2.0.8772/compiler/latest/linux/compiler/lib/intel64_lin:/opt/compiler-explorer/intel-cpp-2022.2.0.8772/compiler/latest/linux/compiler/lib/ia32_lin
+compiler.icc202170.semver=2021.7.0
+compiler.icc202170.options=-gxx-name=/opt/compiler-explorer/gcc-10.1.0/bin/g++
```

Comment:
Do you really want to depend on this old gcc?

Reply:
> Do you really want to depend on this old gcc?

There are 3 options:
* keep everything at 10.2
* new releases use the latest available in CE, old ones do not change
* change old and new releases to use latest available gcc

For me, gcc version matters most because it provides the standard library implementation. Users can override by providing command line arguments. When there are multiple libraries specified, the last one wins. What do you think the default should be?

Also, this is not yaml, is there some macro substitution possible so I do not need to repeat the same strings so much?

Reply:
I think we should not change existing compilers unless the setting is known broken (we want existing links to keep their current behavior).

For new compiler, we should probably use the latest version that makes sense for your compiler. CE should always be up to date wrt to upstream gcc release for x86_64.

Sure, users can specify their own path to gcc, but it's a bit hidden so would not rely too much on it.

There's no fancy macro thing here, you'll need to do some copypasta I'm afraid.

Reply:
updated to 12.2.0

Reply:
Thanks all!